examples/react-native-expo/App.js

```diff
@@ -1,175 +1,166 @@
 // import * as Expo from 'expo'
-import React from 'react'
-import {
-  Text,
-  View,
-  AsyncStorage,
-  Image,
-} from 'react-native'
-import Uppy from '@uppy/core'
-import Tus from '@uppy/tus'
-import UppyFilePicker from '@uppy/react-native'
-import FileList from './FileList'
-import PauseResumeButton from './PauseResumeButton'
-import ProgressBar from './ProgressBar'
-import SelectFiles from './SelectFilesButton'
-import getTusFileReader from './tusFileReader'
-
-export default class App extends React.Component {
-  constructor () {
-    super()
-
-    this.state = {
-      progress: 0,
-      total: 0,
-      file: null,
-      uploadURL: null,
-      isFilePickerVisible: false,
-      isPaused: false,
-      uploadStarted: false,
-      uploadComplete: false,
-      info: null,
-      totalProgress: 0,
-    }
-
-    this.isReactNative = (typeof navigator !== 'undefined'
-      && typeof navigator.product === 'string'
-      && navigator.product.toLowerCase() === 'reactnative')
-
-    this.showFilePicker = this.showFilePicker.bind(this)
-    this.hideFilePicker = this.hideFilePicker.bind(this)
-    this.togglePauseResume = this.togglePauseResume.bind(this)
-
-    console.log('Is this React Native?', this.isReactNative)
-    this.uppy = new Uppy({ autoProceed: true, debug: true })
-    this.uppy.use(Tus, {
-      endpoint: 'https://tusd.tusdemo.net/files/',
+import React, { useEffect, useRef, useState } from "react";
+import { Text, View, AsyncStorage, Image } from "react-native";
+import Uppy from "@uppy/core";
+import Tus from "@uppy/tus";
+import UppyFilePicker from "@uppy/react-native";
+import FileList from "./FileList";
+import PauseResumeButton from "./PauseResumeButton";
+import ProgressBar from "./ProgressBar";
+import SelectFiles from "./SelectFilesButton";
+import getTusFileReader from "./tusFileReader";
+
+export default function App() {
+  const [state, _setState] = useState({
+    progress: 0,
+    total: 0,
+    file: null,
+    uploadURL: null,
+    isFilePickerVisible: false,
+    isPaused: false,
+    uploadStarted: false,
+    uploadComplete: false,
+    info: null,
+    totalProgress: 0,
+  });
+
+  const setState = (newState) =>
+    _setState((oldState) => ({ ...oldState, ...newState }));
+
+  const uppy = useRef();
+
+  useEffect(() => {
+    uppy.current = new Uppy({ autoProceed: true, debug: true });
```

Comment:
If we're using hooks, we should use the `useUppy` hook instead.

Reply:
yes definitely, it's up to you how disruptive this first PR needs to be!

As said in the description, I tried to stick to the bare minimum keeping what was there (wich were all class components)

Reply:
> yes definitely, it's up to you to tell me how disruptive this first PR needs to be!

I'd say things that change, such as the refactor of this components to hooks, should be done well. So I'd consider this request part of that. But fixing uploaders for instance, is a different issue